x-pack/plugins/apm/public/plugin.ts

```diff
@@ -393,18 +403,26 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
         { id: 'tutorial', title: apmTutorialTitle, path: '/tutorial' },
       ],
 
-      async mount(appMountParameters: AppMountParameters<unknown>) {
+      mount: async (appMountParameters: AppMountParameters<unknown>) => {
         // Load application bundle and Get start services
         const [{ renderApp }, [coreStart, pluginsStart]] = await Promise.all([
           import('./application'),
           core.getStartServices(),
         ]);
+        const isCloudEnv = !!pluginSetupDeps.cloud?.isCloudEnabled;
+        const isServerlessEnv =
+          pluginSetupDeps.cloud?.isServerlessEnabled || this.isServerlessEnv;
         return renderApp({
           coreStart,
-          pluginsSetup: pluginSetupDeps,
+          pluginsSetup: pluginSetupDeps as ApmPluginSetupDeps,
           appMountParameters,
           config,
-          pluginsStart: pluginsStart as ApmPluginStartDeps,
+          pluginsStart: {
+            ...pluginsStart,
+            isCloudEnv,
+            isServerlessEnv,
+            kibanaVersion: this.kibanaVersion,
```

Comment:
I'm wondering whether these could go into a separate object named `kibanaEnv` or something like that and decouple them from `pluginsStart`. 

The only difference is that in the `renderApp` we would add another property to  `apmPluginContextValue`.

The same goes for what we currently have in the `infra` plugin.  

My concern is that we're adding things that are not related to plugin dependencies in the `pluginsStart` object

Reply:
Thank you for the comments! IMO the Kibana environment data belongs to the `KibanaReactContext` as it's something related to the Kibana environment and it's not part of the plugin ( as it is the same for all plugins and it's something that we get as a dependency - same as the services and it should be shared ) But based on the [comment](https://github.com/elastic/kibana/pull/173758#pullrequestreview-1800192433) this context will be removed so I am not sure if we want to keep the concept of sharing a context (as package/utility) and reuse it. I don't know what will replace the `KibanaReactContext` but it can be a good place to have this environment-specific data independently there.  
So you are right that with the last changes is better to get them out of the `pluginsStart`. I will do that and I think for now we can keep them in the plugin context until we find a better solution.

Reply:
So for both infra and apm now I am passing `kibanaEnvironment` object separately, can you please take a look again :)